src/neuroglancer/widget/segment_set_widget.ts

```diff
@@ -69,79 +83,204 @@ export class SegmentSetWidget extends RefCounted {
     this.updateClearButtonVisibility();
   }
 
+  private anyRootSegments = () => {
+    return this.displayState.rootSegments.size > 0;
+  }
+
+  private anyHiddenRootSegments = () => {
+    return this.displayState.hiddenRootSegments!.size > 0;
+  }
+
   private updateClearButtonVisibility() {
-    let {clearButton} = this;
-    clearButton.style.display = (this.displayState.rootSegments.size > 0) ? '' : 'none';
+    const {clearButton} = this;
+    clearButton.style.display = (this.anyRootSegments() || this.anyHiddenRootSegments()) ? '' : 'none';
   }
 
-  private handleSetChanged(x: Uint64|Uint64[]|null, added: boolean) {
-    this.updateClearButtonVisibility();
-    let {items} = this;
+  private clearItems() {
+    const {itemContainer, clearButton, enabledItems, disabledItems} = this;
+    while (true) {
+      const lastElement = itemContainer.lastElementChild!;
+      if (lastElement === clearButton) {
+        break;
+      }
+      itemContainer.removeChild(lastElement);
+    }
+    enabledItems.clear();
+    disabledItems.clear();
+  }
+
+  private handleEnabledSetChanged(x: Uint64|Uint64[]|null, added: boolean) {
+    const {enabledItems, disabledItems, hiddenRootSegments, anyHiddenRootSegments} = this;
     if (x === null) {
-      // Cleared.
-      let {itemContainer, clearButton} = this;
-      while (true) {
-        let lastElement = itemContainer.lastElementChild!;
-        if (lastElement === clearButton) {
-          break;
+      if (! anyHiddenRootSegments()) {
+        // Cleared.
+        this.clearItems();
+      }
+    } else if (added) {
+      for (const segmentID of Array<Uint64>().concat(x)) {
+        const segmentIDString = segmentID.toString();
+        const disabledItem = disabledItems.get(segmentIDString);
+        // Make sure item not already added
+        if (! disabledItem) {
+          this.addElement(segmentIDString);
         }
-        itemContainer.removeChild(lastElement);
+        else {
+          // Preparing to enable or disable an element
+          enabledItems.set(segmentIDString, disabledItem);
+          hiddenRootSegments!.delete(x);
+          this.setItemsToggleButtonToHideSegment(disabledItem, segmentIDString);
+        }
+      }
+    } else {
+      for (const segmentID of Array<Uint64>().concat(x)) {
+        const segmentIDString = segmentID.toString();
+        // Make sure item has been deleted, instead of disabled
+        if (! disabledItems.get(segmentIDString)) {
+          let itemElement = enabledItems.get(segmentIDString)!;
+          itemElement.parentElement!.removeChild(itemElement);
+        }
+        enabledItems.delete(segmentIDString);
+      }
+    }
+  }
+
+  private handleDisabledSetChanged(x: Uint64|Uint64[]|null, added: boolean) {
```

Comment:
Missing the call to `this.updateClearButtonVisibility();` here to check if the `clear-button` is still required